### ISSUE TITLE
`@schema_columns_hash` is not cleared by `.reset_column_information`.

### ISSUE THREAD

**Issue by koshigoe** [zdennis/activerecord-import#809]
https://github.com/zdennis/activerecord-import/blob/0fc4ef1f4990e8338f78b5bd5aa713ad22622de1/lib/activerecord-import/import.rb#L971-L977

This change introduced by #799.

- see https://github.com/zdennis/activerecord-import/pull/799/files#diff-430ef1d0eb08958d694bf47c24a5f931efc301113c3f80a6b8c70a7aa2603209R967-R970

Before #799, activerecord-import use `columns_hash` that provieded by ActiveRedord.
After #799, activerecord-import use `schema_columns_hash` instead of `columns_hash`.
The `schema_columns_hash` memoized but reset by `.reset_column_information`.

So exception `ActiveRecord::Import::MissingColumnError` occur when change schema dynamically.

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "activerecord", "~> 7.0.7"
  gem "activerecord-import", ENV.fetch('AR_IMPORT_VERSION', '1.4.1')
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: 'test.sqlite3')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema[7.0].define do
  create_table :tests, force: true do |t|
    t.string :column_1
  end
end

class Test < ActiveRecord::Base
end

class AddColumnToTest < ActiveRecord::Migration[7.0]
  def change
    add_column :tests, :column_2, :string
  end
end

class BugTest < Minitest::Test
  def test_import_after_add_column
    # cache @schema_columns_hash
    Test.import(%w(column_1), [%w(a)])

    # add column
    AddColumnToTest.migrate(:up)

    # reset cached schema information
    Test.reset_column_information

    # raise ActiveRecord::Import::MissingColumnError on activerecord-import v1.5.0
    Test.import(%w(column_1 column_2), [%w(a b)])
    assert_equal([["a", nil], ["a", "b"]], Test.order(:id).pluck(:column_1, :column_2))
  end
end
```

**Comment by jkowens** [comment]
Thanks @koshigoe could you provide a PR for this issue?

**Comment by koshigoe** [comment]
@jkowens 
I'm sorry, but I can't make time for a while.

**Comment by jkowens** [comment]
@koshigoe I wonder if we can just stop memoizing  `@schema_columns_hash`. I'm not sure what kind of performance toll that will take.

@leonidkroka any thoughts?

**Comment by leonidkroka** [comment]
@jkowens yeah, stopping memoizing `@schema_columns_hash` seems to be the best solution

It was done in a similar way to the ActiveRecord method `columns`:
```
def columns
  load_schema
  @columns ||= columns_hash.values.freeze
end
```
however `reset_column_information` does:
```
connection.clear_cache!
...
reload_schema_from_cache # @columns_hash = nil
```
Seems I didn't take care of that

--------------------------------------------
> not sure what kind of performance toll that will take
Shouldn't take a performance downgrade as long as `ActiveRecord::ConnectionAdapters::SchemaCache` is memoizing `columns_hash`.

**Comment by Pritilender** [comment]
@jkowens is there a plan to make a release of the gem with this fix included? I don't see it on the current v1.5 gem and I'm facing issues related to this memoization.

**Comment by jkowens** [comment]
@Pritilender I apologize, I forgot this hadn't been released. I should probably be able to cut a release by tomorrow.

**Comment by Pritilender** [comment]
@jkowens thank you! Appreciate it!

**Comment by jkowens** [comment]
Fix released in v1.5.1 👍
